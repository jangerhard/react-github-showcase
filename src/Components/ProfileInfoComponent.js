import PropTypes from 'prop-types';
import React from 'react';
import {
    TextBlock,
    MediaBlock,
    TextRow,
    RectShape,
    RoundShape
} from 'react-placeholder/lib/placeholders';
import ReactPlaceholder from "react-placeholder";

const styles = {
    display: "flex",
    flexDirection: "column",
    width: "10em",
    marginRight: "1.5em",
    textAlign: "center"
};

const imageStyles = {
    borderRadius: "50%"
};

const awesomePlaceholder =
    <div className="profileInfo" style={styles}>
        <RoundShape color='grey' style={{width: 100, height: 100}}/>
        <TextBlock rows={1} color='grey'/>
    </div>;

const ProfileInfoComponent = ({avatarUrl, fullName, loading}) =>
    <ReactPlaceholder ready={!loading}
                      customPlaceholder={awesomePlaceholder}>
        <div className="profileInfo" style={styles}>
            <img className="pic" style={imageStyles} src={avatarUrl}/>

            {fullName}
        </div>
    </ReactPlaceholder>;

ProfileInfoComponent.propTypes = {
    fullName: PropTypes.string.isRequired,
    avatarUrl: PropTypes.string.isRequired
};

export default ProfileInfoComponent;