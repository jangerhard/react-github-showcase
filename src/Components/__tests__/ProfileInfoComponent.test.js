import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import ProfileInfoComponent from "../ProfileInfoComponent";

const fullName = 'Test Testington';
const avatarUrl = 'www.avatarurl.com';

it('renders profile-info correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <ProfileInfoComponent
            fullName={fullName}
            avatarUrl={avatarUrl}
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
});
it('renders profile-info correctly when loading', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <ProfileInfoComponent
            fullName={fullName}
            avatarUrl={avatarUrl}
            loading
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
});