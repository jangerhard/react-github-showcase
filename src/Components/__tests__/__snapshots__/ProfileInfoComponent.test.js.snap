// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders profile-info correctly 1`] = `
<div
  className="profileInfo"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "marginRight": "1.5em",
      "textAlign": "center",
      "width": "10em",
    }
  }
>
  <img
    className="pic"
    src="www.avatarurl.com"
    style={
      Object {
        "borderRadius": "50%",
      }
    }
  />
  Test Testington
</div>
`;
